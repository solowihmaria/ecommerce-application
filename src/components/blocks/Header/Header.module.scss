.header {
    display: flex;
    justify-content: space-between;
    padding: 20px 30px 20px 20px;
    background-color: #5e8567;
}

.menu {
    display: flex;
    gap: 10px;
}
