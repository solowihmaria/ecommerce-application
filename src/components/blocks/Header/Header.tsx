import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { Button } from '../../ui/Button';
import styles from './Header.module.scss';

export const Header = () => {
    const [isLogined, setIsLogined] = useState(true);

    return (
        <nav className={styles.header}>
            <div className={styles.logoContainer}>
                <Link to="/main">
                    <Button>Main</Button>
                </Link>
            </div>
            <div className={styles.menu}>
                {isLogined ? (
                    <>
                        <Link to="/login">
                            <Button onClick={() => setIsLogined(false)}>
                                Logout
                            </Button>
                        </Link>
                    </>
                ) : (
                    <>
                        <Link to="/login">
                            <Button>Sign In</Button>
                        </Link>
                        <Link to="/register">
                            <Button>Sign Up</Button>
                        </Link>
                    </>
                )}
            </div>
        </nav>
    );
};
