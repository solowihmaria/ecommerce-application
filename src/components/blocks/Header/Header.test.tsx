import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import { Header } from './Header';

test('Render Header component', () => {
    render(
        <MemoryRouter>
            <Header />
        </MemoryRouter>
    );
    const header = screen.getByTestId('test-id-header');
    expect(header).toBeInTheDocument();
    expect(header).toMatchSnapshot();
});
