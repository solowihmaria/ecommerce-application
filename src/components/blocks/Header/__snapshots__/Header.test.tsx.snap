// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header component 1`] = `
<header
  class="headerWrapper"
  data-testid="test-id-header"
>
  <nav
    class="header"
  >
    <div
      class="leftSection"
    >
      <a
        class="logoLink"
        data-discover="true"
        href="/main"
      >
        <svg
          data-testid="mock-svg"
        />
      </a>
    </div>
    <div
      class="centerSection"
    >
      <a
        data-discover="true"
        href="/main"
      >
        <button
          class="button ghost navButton"
        >
          Main
        </button>
      </a>
      <a
        data-discover="true"
        href="/catalog"
      >
        <button
          class="button ghost navButton"
        >
          Catalog
        </button>
      </a>
      <a
        data-discover="true"
        href="/"
      >
        <button
          class="button ghost navButton"
        >
          About
        </button>
      </a>
    </div>
    <div
      class="rightSection"
    >
      <svg
        class="profileIcon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M11 7 9.6 8.4l2.6 2.6H2v2h10.2l-2.6 2.6L11 17l5-5-5-5zm9 12h-8v2h8c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-8v2h8v14z"
        />
      </svg>
      <div
        class="profileMenu"
      >
        <a
          class="link"
          data-discover="true"
          href="/login"
        >
          <button
            class="button ghost authButton"
          >
            Sign In
          </button>
        </a>
        <a
          class="link"
          data-discover="true"
          href="/register"
        >
          <button
            class="button ghost authButton"
          >
            Sign Up
          </button>
        </a>
      </div>
      <a
        class="cartLink"
        data-discover="true"
        href="/"
      >
        <svg
          data-testid="mock-svg"
        />
      </a>
      <div
        class="hamburger"
      >
        <svg
          class="hamburgerIcon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </div>
    </div>
  </nav>
</header>
`;
